### PR TITLE
Tagged Clustering.jl v0.4

### PR DESCRIPTION
**Pull request by slundberg**
Bumps the version of clustering to include hclust.

### CODE REVIEW

**Comment by johnmyleswhite** [comment]
LGTM

File: Clustering/versions/0.4.0/requires [hunk 1]
```diff
@@ -0,0 +1,4 @@
+julia 0.3-
```
Comment (tkelman) [inline Clustering/versions/0.4.0/requires]: unless you're testing with a prerelease, this should drop the trailing `-`

Comment (slundberg) [inline Clustering/versions/0.4.0/requires]: I just copied how things are in the current 0.3.3 version. If you want me
to change it I can.

Comment (milktrader) [inline Clustering/versions/0.4.0/requires]: Yeah, the dash cruft is everywhere! We're just trying to mop it up :smile:

**Pull request merged**